insert of german chatter texts into text_loc3

Raz0r1337 opened the pull request:
The original files from the base directory have been overwritten somewhere, and therefore they can be removed from there as well.

If possible revert #961

hermensbas commented:
We cant just revert #961, you need to add revert and solve merge-conflicts or remove it manually.

Raz0r1337 commented:
If I do this for German, should I also remove all other languages except English from the base file and prepare update sqls for them? That would clean up the base file a bit.

Raz0r1337 commented:
Am I doing something wrong here? The structure is exactly the same as all the other language files, and when I run the SQL file, it executes without errors, but nothing is written to the database.

Raz0r1337 commented:
Sorry, I'm a bit tired; I overlooked that I was inserting the text into the wrong column (I should have used "text_loc3" instead of "text").

It's fixed now.

hermensbas commented:
this might be sufficient; overwriting base script in combination with german update script should end up in valid state again.

hermensbas commented:
Can someone else approve? So we can extend of playerbase with german players? :)

noisiver reviewed (dismissed):
I don't speak german and can read only a little but I'll approve this since I trust @hermensbas and it appears to look good code-wise

It0w commented:
Thanks for the translation :)

I think the following IDs have errors in their variable names.

```
178
187
190
198
201
205
206
207
225
230
```

Some of the translations are a bit strange, but for MTL, considering the goal is to stay as close to the original as possible, they're pretty good.

It0w commented:
@kadeshar
I checked my SQL table, and it turns out that my IDs don't match either.
Are the IDs assigned randomly when the table is created? I looked through the SQL file (```data/sql/playerbots/base/ai_playerbot_texts.sql```), but I couldn't find any IDs.

If the IDs are different for each instance, it would probably be better to refer to the text field rather than the IDs for match and replace.
Another option is to create the base table with uniform IDs so that they can be reused later. 

> I translated all English chatter texts 1-1 and only adjusted minor details that didn't make any sense in German. I would have liked to have taken a more creative approach, but then the PR would have been rejected for not being close enough to the original.

I know so good work :)

kadeshar commented:
About records difference i only applied creation scripts. I didnt reapplied updates which adds new rows.

Raz0r1337 commented:
Great, so that means all localization updates that are already in there and those that are still pending as PRs are useless because nothing is sorted correctly except for the ones that were created because the IDs match there.

I don't think it's possible to sort it using the original English text, because there are a few duplicates in the texts that make it impossible to uniquely identify a line, so SQL will just spit out errors.

The most sensible thing to do would be to assign unique IDs to the entries in the base file and then have the PR owners realign the translations with them.

Raz0r1337 commented:
So my ai_playerbot_texts database has exactly 858 lines, no more, no less.
I had everything popularized from the core and all updates installed. I don't know where the over 1700 lines come from.

Or do you mean the basefile in the source code?
There are still many comments and further instructions for the db in there, but they don't play a role in the translations.

Raz0r1337 commented:
I've read the entrys that are shown on your screenshot, the English texts are matching the Germans now.

kadeshar commented:
Yeap but that is existing script which you cleaning up in this PR

Raz0r1337 commented:
I had a look in the update folder and looked at the two translations that have already been installed. All translations work with the IDs as mapping and they all have about 850 lines. For my database, I loaded both the original and the base file from my pr and loaded the core and had the DB filled in two cloned VMs. I always end up with 853 lines and all translations fit perfectly as far as I can see. My test VM is just the current PB fork of AzerothCore and mod-playerbots.

I'm not deep enough in the matter to be able to say more precisely. Do you have any idea what's going wrong here?

kadeshar commented:
I see reason 
<img width="879" height="515" alt="obraz" src="https://github.com/user-attachments/assets/5dfaaed5-bf02-4e11-a4df-097ddb239198" />

900 lines was applied to base file but update script wasnt created. Databases created before this commit have less records.

We need to:

- modify base file to include id`s in INSERT statement
<img width="1420" height="245" alt="obraz" src="https://github.com/user-attachments/assets/d29e3b89-48bd-4c77-8494-d9aa184420b4" />

- create update script containing whole base file

- personally im think that german translation should be included in base and update script

If you will take care that let me know. If not, i will create issue or fix that myself.

Raz0r1337 commented:
Thank you for looking at this.

I can insert the ids, but I don't think I can adapt my translation so that the IDs fit. And I don't think I can integrate my translation into the base file that the IDs fit.

Raz0r1337 commented:
I am currently correcting and completing the German translation of the chatter texts.

I have a question about placeholders such as %item_link. I have already noticed that a - directly in front or back of the placeholder ensures that it is not replaced but written out in full.

My question is: does this apply to all symbols that appear directly before or after a placeholder? For example: %item_link, is BiS for hunters or an ? after the placeholders.

Raz0r1337 commented:
@kadeshar

I took another week and extracted all the original English strings again and had them machine-translated into German. Then I went through every single line several times and adjusted everything so that it really makes sense for us Germans.

Now everything is fine with the localization.

Raz0r1337 commented:
This was just a rename, i couldnt find any option on the github website to rename the file to fix a typo in the name.

kadeshar commented:
Thanks for effort :) Now german language should be great. Restore default base file. Any modification of this file doesnt have sense because is entire override in update script.
Also update script update id i dont know why. Update should look like 
```
UPDATE `ai_playerbot_texts` SET `text_loc3`='am Arsch der Welt' WHERE `id`=1;
```

kadeshar commented:
Update script should contain only rows which are real updated.

icemansparks commented:
> Update script should contain only rows which are real updated.

But since some of the german entries had been added to the base file some time ago (therefore they're missing in existing installations from before that) the proper solution now would be to all have them in the update file - wouldn't it?

Raz0r1337 commented:
Again, just to clarify why I find it ok to replace all in the first time.

I had all the English texts from the DB that I had previously updated from the core fully updated. I first had these texts translated automatically with a script using Deepl. Then I went through each line twice manually and corrected everything that didn't fit in German and wasn't well formulated until it sounded good and made sense. If lines correspond to the text in the base file, it's more or less a coincidence. Since this is supposed to be a complete localisation, it makes sense to translate all lines once, especially since there was no German in the original base file and I had added it there. However, that was a purely machine-generated translation from chatGPT3, and now it is a localisation made fully by humans. Any further changes, if anything else needs to be fixed, I will only change the respective line, but as an initial translation, this makes sense.

@kadeshar , can you please show me which line(s) you found that differs from the base file? I'll take another look at it today and that would speed up the process.

kadeshar commented:
PR still contains changes in ai_playerbot_texts.sql. This file should not be touched because whole its content is override by update script.
